test/decl/typealias/protocol.swift

```diff
@@ -268,7 +268,7 @@ extension P10 {
   typealias U = Float
 }
 
-extension P10 where T == Int { } // expected-warning{{neither type in same-type constraint ('Self.T' (aka 'Int') or 'Int') refers to a generic parameter or associated type}}
+extension P10 where T == Int { } // expected-warning{{neither type in same-type constraint ('Int' or 'Int') refers to a generic parameter or associated type}}
```

Comment:
How did we end up losing sugar here?

Reply:
This is from the change to GSB::expandConformanceRequirement() which replaces the SubstitutionMap usage (which does resugaring) with the simpler substitution. I did that to avoid a request cycle that's surfaced by another change.

For various reasons the requirement machine doesn't preserve sugar in where clauses at all, so I'm not too worried about it.